CycloneDX/Services/ProjectAssetsFileService.cs

```diff
@@ -95,51 +92,13 @@ public HashSet<NugetPackage> GetNugetPackages(string projectFilePath, string pro
 
             return packages;
         }
-        public bool SetIsDirectReference(string packageName, string targetRuntime, JsonElement jsonContent)
+        public bool SetIsDirectReference(LibraryDependency dependency)
         {
-            string framework = TargetFrameworkToAlias(targetRuntime);
-            JsonElement packageProperties;
-            if (jsonContent.GetProperty(framework).GetProperty("dependencies").TryGetProperty(packageName, out packageProperties))
-            {
-                // every direct reference has target property
-                return packageProperties.TryGetProperty("target", out _);
-            }
-            return false;
+            return dependency?.ReferenceType == LibraryDependencyReferenceType.Direct;
         }
-        public bool SetIsDevDependency(string packageName, string targetRuntime, JsonElement jsonContent)
+        public bool SetIsDevDependency(LibraryDependency dependency)
         {
-            string framework = TargetFrameworkToAlias(targetRuntime);
-            JsonElement packageProperties;
-            if (jsonContent.GetProperty(framework).GetProperty("dependencies").TryGetProperty(packageName, out packageProperties))
-            {
-                // suppressParent: exists only for development dependencies
-                return packageProperties.TryGetProperty("suppressParent", out _);
-            }
-            return false;
-        }
-
-        /// <summary>
-        /// Converts an asset file's target framework value into a csproj target framework value.
-        ///
-        /// Examples:
-        ///     .NetStandard,Version=V3.1 => netstandard3.1
-        ///     .NetCoreApp,Version=V3.1  => netcoreapp3.1
-        ///     netcoreapp3.1  => netcoreapp3.1
-        ///     net6.0  => net6.0
-        /// </summary>
-        private string TargetFrameworkToAlias(string target)
-        {
-            if (!string.IsNullOrEmpty(target))
-            {
-                target = target.ToLowerInvariant().TrimStart('.');
-                var targetParts = target.Split(",version=v");
-                if (targetParts.Length == 2)
-                {
-                    return string.Join("", targetParts);
-                }
-                return targetParts[0];
-            }
-            return null;
+            return dependency != null && dependency.SuppressParent != LibraryIncludeFlagUtils.DefaultSuppressParent;
```

Comment:
The existance of `suppressParent` is not a good metric for checking if it's a development dependency, as it CAN be set in other occasions as well. The existing code would have failed if `PrivateAssets="Build;Analyzers"` was set on the `PackageReference` (keeping `ContentFiles`).

Reply:
Technically, the ones interesting here would be `Compile`, `Runtime` and `Native`. If any of these would NOT be suppressed, at least some files from the package WOULD end up in the deployable. `DefaultSuppressParent` contains  `Build;ContentFiles;Analyzers` are marked private assets (see https://learn.microsoft.com/en-us/nuget/consume-packages/package-references-in-project-files#controlling-dependency-assets and https://github.com/NuGet/NuGet.Client/blob/6.7.0.110/src/NuGet.Core/NuGet.LibraryModel/LibraryIncludeFlagUtils.cs#L16-L20 for reference).